### DISCUSSION TITLE
Complete the auth state change implementation

### DISCUSSION THREAD

**Discussion by wiverson** [supabase/gotrue-csharp#55] [General]
Right now I'm not getting proper auth state changes for several code paths (e.g. sign in with apple, but also for some email paths, etc).

It looks like the auth state stuff is a bit incomplete, ok to add a bunch of tests and beef this stuff up?

LMK thoughts, ok to send PR?

**Comment by acupofjose** [comment]
Oh interesting. Yeah please do!

**Comment by wiverson** [comment]
Resolved over several PRs including https://github.com/supabase-community/gotrue-csharp/pull/57

Woot!
